src/overlays/actors/ovl_Demo_Effect/z_demo_effect.h

```diff
@@ -6,11 +6,134 @@
 
 struct DemoEffect;
 
+typedef void (*DemoEffectFunc)(struct DemoEffect* this, GlobalContext* globalCtx);
+
 typedef struct DemoEffect {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x54];
+    /* 0x014C */ SkelAnimeCurve skelCurve;
+    /* 0x016C */ u8 initObjectBankIndex;
+    /* 0x0170 */ Gfx* jewelDisplayList;
+    /* 0x0174 */ Gfx* jewelHolderDisplayList;
+    /* 0x0178 */ u8 primXluColor[3];
+    /* 0x017B */ u8 envXluColor[3];
+    /* 0x017E */ u8 primOpaColor[3];
+    /* 0x0181 */ u8 envOpaColor[3];
+    /* 0x0184 */ union {
+        struct {
+            u8 timer;
+        } fireBall;
+        struct {
+            u8 opacity;
+            u8 scale;
+            u8 pad;
+            s16 rotation;
+        } blueOrb;
+        struct {
+            u8 opacity;
+            u8 scaleFlag;
+            u8 flicker;
+            s16 rotation;
+        } lightEffect;
+        struct {
+            u8 opacity;
+        } lgtShower;
+        struct {
+            u8 type;
+            u8 lightRingSpawnDelay;
+            u8 rotation;
+            s16 lightRingSpawnTimer;
+        } godLgt;
+        struct {
+            u8 timerIncrement;
+            u8 opacity;
+            u8 pad;
+            s16 timer;
+        } lightRing;
+        struct {
+            u8 triforceSpotOpacity;
+            u8 lightColumnOpacity;
+            u8 crystalLightOpacity;
+            s16 rotation;
+        } triforceSpot;
+        struct {
+            u8 isPositionInit;
+            u8 isLoaded;
+            u8 giIndex;
+            s16 rotation;
+        } getItem;
+        struct {
+            u8 pad;
+            u8 pad2;
+            u8 pad3;
+            s16 shrinkTimer;
+        } timeWarp;
+        struct {
+            u8 type;
+            u8 isPositionInit;
+            u8 opacity;
+            s16 timer;
+        } jewel;
+        struct {
+            u8 timer;
+        } dust;
+    };
+    /* 0x018A */ s16 effectFlags;
+    /* 0x018C */ s16 csActionId;
+    /* 0x018E */ Vec3s jewelCsRotation;
+    /* 0x0194 */ DemoEffectFunc initUpdateFunc;
+    /* 0x0198 */ DemoEffectFunc initDrawFunc;
+    /* 0x019C */ DemoEffectFunc updateFunc;
 } DemoEffect; // size = 0x01A0
 
-extern const ActorInit Demo_Effect_InitVars;
+// These names come from the objects that correspond to this actor type.
+typedef enum Demo_Effect_Effect_Type {
```

Comment:
I believe enum entries should be all caps.

Reply:
Let's let Fig and/or Roman comment here then. :)

Reply:
I agree enum names should be in all caps, and I also am not a fan of "Effect_Effect". I know you didn't ask for my opinion but yee